src/main/java/eu/darkbot/logic/modules/MapModule.java

```diff
@@ -0,0 +1,138 @@
+package eu.darkbot.logic.modules;
+
+import com.github.manolo8.darkbot.utils.I18n;
+import eu.darkbot.api.PluginAPI;
+import eu.darkbot.api.entities.Portal;
+import eu.darkbot.api.entities.utils.Map;
+import eu.darkbot.api.managers.EntitiesAPI;
+import eu.darkbot.api.managers.HeroAPI;
+import eu.darkbot.api.managers.MovementAPI;
+import eu.darkbot.api.managers.PetAPI;
+import eu.darkbot.api.managers.StarAPI;
+import eu.darkbot.api.objects.Location;
+import eu.darkbot.api.utils.Listener;
+import eu.darkbot.logic.utils.PortalJumper;
+
+import java.util.Collection;
+
+public class MapModule extends TemporalModule {
+
+    private Traveler traveler;
+
+    public MapModule(PluginAPI api, Map target) {
+        super(api);
+        if (traveler == null)
+            this.traveler = new Traveler(api);
+
+        this.traveler.setTarget(target);
+    }
+
+    public void setTarget(Map target) {
+        this.traveler.setTarget(target);
+    }
+
+    @Override
+    public String getStatus() {
+        return  traveler.current != null ?
+                I18n.get("module.map_travel.status.has_next", traveler.target.getName(),
+                        traveler.current.getTargetMap().map(Map::getName).orElse("unknown?")) :
+                I18n.get("module.map_travel.status.no_next", traveler.target.getName());
+    }
+
+    @Override
+    public void onTickModule() {
+        if (!traveler.isDone()) traveler.tick();
+        if (traveler.isDone()) goBack();
+    }
+
+    private static class Traveler {
```

Comment:
Whis is this traveler embedded inside map module?
Sometimes other parts of the bot may want to travel without necessarily handing over the full control to the map module, the safety finder is an example of this, where it uses a traveler to go back to the previous map